src/main/play-26/uk/gov/hmrc/play/HeaderCarrierConverter.scala

```diff
@@ -38,7 +39,13 @@ trait HeaderCarrierConverter {
     session: Option[Session]       = None,
     request: Option[RequestHeader] = None): HeaderCarrier =
     session.fold(fromHeaders(headers, request)) {
-      val cookies = Cookies.fromCookieHeader(headers.get(play.api.http.HeaderNames.COOKIE))
+      // Cookie setting changed between Play 2.5 and Play 2.6, this now checks both ways
+      // cookie can be set for backwards compatibility
+      val cookiesInHeader =
+        Cookies.fromCookieHeader(headers.get(PlayHeaderNames.COOKIE)).toList
+      val cookiesInSession =
+        request.map(_.cookies).map(_.toList).getOrElse(List.empty)
```

Comment:
If you click through from `request.map(_.cookies)` you get this:

    lazy val cookies: Cookies = Cookies.fromCookieHeader(headers.get(play.api.http.HeaderNames.COOKIE))

Am I missing something? The two methods of getting cookies look identical?

Reply:
Ah yes, I am... that is in Play25 sources. The Play26 stuff uses the typed attrs!